homeassistant/components/media_player/squeezebox.py

```diff
@@ -203,9 +191,30 @@ def state(self):
                 return STATE_IDLE
         return STATE_UNKNOWN
 
-    def update(self):
-        """Retrieve latest state."""
-        self._status = self._lms.get_player_status(self._id)
+    @asyncio.coroutine
+    def async_query(self, *parameters):
+        """Send a command to the LMS."""
+        response = yield from self._lms.async_query(
+            *parameters, player=self._id)
+        return response
+
+    def query(self, *parameters):
+        """Queue up a command to send the LMS."""
+        self.hass.loop.create_task(self.async_query(*parameters))
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Retrieve the current state of the player."""
+        tags = 'adKl'
+        response = yield from self.async_query(
+            "status", "-", "1", "tags:{tags}"
+            .format(tags=tags))
+
+        try:
+            self._status = response.copy()
+            self._status.update(response["remoteMeta"])
+        except KeyError:
+            self._status = response.copy()
```

Comment:
Can this just be a `pass`? If the second line fails _status should still have a response copy.